clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/DeleteCloudFormationChangeSetAtomicOperation.java

```
@@ -0,0 +1,70 @@
+ */
+package com.netflix.spinnaker.clouddriver.aws.deploy.ops;
+
+import com.amazonaws.services.cloudformation.AmazonCloudFormation;
+import com.amazonaws.services.cloudformation.model.*;
+import com.netflix.spinnaker.clouddriver.aws.deploy.description.DeleteCloudFormationChangeSetDescription;
+import com.netflix.spinnaker.clouddriver.aws.security.AmazonClientProvider;
+import com.netflix.spinnaker.clouddriver.data.task.Task;
+import com.netflix.spinnaker.clouddriver.data.task.TaskRepository;
+import com.netflix.spinnaker.clouddriver.orchestration.AtomicOperation;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+
+@Slf4j
+public class DeleteCloudFormationChangeSetAtomicOperation implements AtomicOperation<Map> {
+
+  private static final String BASE_PHASE = "DELETE_CLOUDFORMATION_CHANGESET";
+
+  @Autowired AmazonClientProvider amazonClientProvider;
+
+  private DeleteCloudFormationChangeSetDescription description;
+
+  public DeleteCloudFormationChangeSetAtomicOperation(
+      DeleteCloudFormationChangeSetDescription deployCloudFormationDescription) {
+    this.description = deployCloudFormationDescription;
+  }
+
+  @Override
+  public Map operate(List priorOutputs) {
+    Task task = TaskRepository.threadLocalTask.get();
+    AmazonCloudFormation amazonCloudFormation =
+        amazonClientProvider.getAmazonCloudFormation(
+            description.getCredentials(), description.getRegion());
+
+    DeleteChangeSetRequest deleteChangeSetRequest =
+        new DeleteChangeSetRequest()
+            .withStackName(description.getStackName())
+            .withChangeSetName(description.getChangeSetName());
+    try {
+      task.updateStatus(BASE_PHASE, "Deleting CloudFormation ChangeSet");
+      amazonCloudFormation.deleteChangeSet(deleteChangeSetRequest);
+    } catch (AmazonCloudFormationException e) {
+      log.error(
```

is it OK that this error is ignored?

---

Do you mean not returning any error back to orca? Fair comment, I'll return an error so orca can handle it appropiately, probably with a retryable task :+1:

---

Done in https://github.com/spinnaker/clouddriver/pull/4101/commits/c7cd1349d81479884881e3937a0e497146128f0d